aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -613,6 +606,36 @@ Tensor& linalg_matrix_rank_out(
   return result;
 }
 
+Tensor get_matrix_rank_result_tensor(const Tensor& input) {
```

Comment:
This might be more of a question for the linalg folks, but I'm confused: linalg.matrix_rank is differentiable, right?

Autograd doesn't support the `sum_out` operation. So how does autograd support for matrix_rank work?

The reason why I am asking is because it looks like we won't be able to use vmap over this (for the same reason -- vmap doesn't support out= variants).

Reply:
`matrix_rank` outputs a tensor of integers, so it's not differentiable.

Reply:
Okay I answered my question: matrix_rank is not differentiable :)